lib/simplecov/filter.rb

```diff
@@ -63,11 +65,23 @@ def matches?(source_file)
   end
 
   class ArrayFilter < SimpleCov::Filter
-    # Returns true if any of the file paths passed in the given array matches the string
-    # configured when initializing this Filter with StringFilter.new(['some/path', 'other/path'])
+    def initialize(filter_argument)
+      filter_argument.map! do |arg|
+        if arg.is_a?(SimpleCov::Filter)
+          arg
+        else
+          Filter.class_for_argument(arg).new(arg)
+        end
+      end
+
+      super(filter_argument)
+    end
```

Comment:
There is definitely a hole in the specs. I had to do manual tests to make sure this worked. I'll see if I can whip up some specs.